rfd/0133-connect-my-computer.md

```diff
@@ -0,0 +1,621 @@
+the `will-download` event of `webContents.session`. We should encapsulate those APIs within a single
+function to avoid the downloading logic being spread across files and listeners.
+
+We have to be mindful about keeping the preload APIs as constrained as possible. The renderer should
+not get the ability to download an arbitrary file to an arbitrary path. If possible, the download
+process should be totally encapsulated from the renderer, with the renderer merely signaling the
+intent to start the agent and the main process handling everything else.
+
+##### Storing the node agent
+
+The binary is stored in `~/Library/Caches/Teleport Connect` on macOS and `~/.cache/Teleport Connect`
+on Linux. It is extracted to a directory of the same name as the downloaded tar.gz archive, for
+example `teleport-v13.0.4-darwin-arm64-bin`. As the parent directory can be cleared by the OS or the
+user, before each launch Connect is going to check if the agent exists and download it automatically
+if not.
+
+So far we’ve been storing all data in Electron’s `userData` which is `~/Library/Application Support`
+on macOS. [That directory is not suited for storing large files](https://www.electronjs.org/docs/latest/api/app#appgetpathname) as it might be backed up to
+cloud storage. 
+
+On top of that, the agent binary is something that can be easily recreated if needed by downloading
+it again. After reading Apple Developer docs ([Optimizing Your App’s Data for iCloud Backup](https://developer.apple.com/documentation/foundation/optimizing_your_app_s_data_for_icloud_backup/#3928528)
+and archived [File System Programming Guide](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW4)),
+we’ve decided that `~/Library/Caches` is a suitable directory for storing the agent.
+
+Out of the box, Electron doesn’t expose `~/Library/Caches` through its `app.getPath` function,
+unlike `~/Library/Application Support` which is retrievable by calling `app.getPath('userData')`.
+
+A good candidate for that would be `sessionData` which the docs describe as “the directory for
+storing data generated by Session, such as localStorage, cookies, disk cache, downloaded
+dictionaries, network state, devtools files.” Unfortunately, for legacy reasons `sessionData` points
+to the same directory as `userData`. [The docs recommend changing the location](https://www.electronjs.org/docs/latest/api/app#appgetpathname) of this directory to
+another location.
+
+There’s [an outstanding issue about Electron storing its caches in the wrong location](https://github.com/electron/electron/issues/8124). Based on the Apple Developer docs and
+the discussion under that issue, changing the location of `sessionData` to `~/Library/Caches` on
+macOS and `~/.cache` on Linux seems like a reasonable thing to do. As we have not been using cookies
```

Comment:
As per https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html , we should use `$XDG_CACHE_HOME` on Linux, defaulting to `~/.cache` if unset.